marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -586,20 +582,13 @@ function findFlowAndStepNameFromProvenanceRecords(provenanceRecord, stepNames, s
       associatedWith[associatedWith.length-1] = artifactTobeRemoved;
       break;
     }
-    associatedWith.pop();
   }
-  console.log("associatedWith array after finding flowName: " + associatedWith);
+  associatedWith.pop();
```

Comment:
Is this the fix then - that the pop() call was occurring within the for loop?

Reply:
If the associatedWith was something like this ["inline", "map", "entity-services-mapping"], then the test runs fine.
The test fails when the associatedWith is something like this ["entity-services-mapping", "inline", "map"].
The pop should happen after breaking out of the for loop.